services/venice-server/src/main/java/com/linkedin/venice/stats/ServerHttpRequestStats.java

```diff
@@ -295,23 +295,33 @@ public ServerHttpRequestStats(
         () -> totalStats.earlyTerminatedEarlyRequestCountSensor,
         new OccurrenceRate());
 
-    if (isKeyValueProfilingEnabled) {
-      // size profiling is expensive
+    if (isKeyValueProfilingEnabled || requestType == RequestType.SINGLE_GET) {
```

Comment:
Should we introduce a separate flag specifically for tracking percentiles?

With the current change, percentiles for single-gets won’t be tracked unless isKeyValueProfilingEnabled is enabled. Additionally, enabling isKeyValueProfilingEnabled automatically tracks key-value sizes for all request types. It might be worth considering separating these two functionalities with separate config options?.

Reply:
OTHOH this might be fine if, by default, we want to track only average and maximum sizes for single-gets. If more detailed data is needed, such as percentiles, users can enable key-value profiling explicitly.

Reply:
I don't think we need to split the two functionalities. We can add it later if we need it

Reply:
Sounds good.